tensordict/prototype/tensorclass.py

```diff
@@ -302,16 +343,32 @@ def _setitem(self, item, value):
         raise ValueError("Invalid indexing arguments.")
     if not isinstance(value, self.__class__):
         raise ValueError("__setitem__ is only allowed for same-class assignement")
+    for key, val in self.non_tensordict.items():
+        if not isinstance(val, _accepted_classes):
+            if val and val != value.non_tensordict[key]:
```

Comment:
this would fail in case of a nestedclass

Reply:
you marked this as resolved but I don't see any change in the code

Reply:
Changed the code from before. Nested case is handled above with if and the else only has non_tensor data